scienceworld/scienceworld.py

```diff
@@ -121,9 +120,13 @@ def reset(self):
 
     # Simplifications
     def getSimplificationsUsed(self):
+        deprecated_api_warning(logger, True, True)
+
         return self.server.getSimplificationsUsed()
```

Comment:
I would suggest using [`warnings.warn`](https://docs.python.org/3.12/library/warnings.html#warnings.warn) instead and directly throw a [`DeprecationWarning`](https://docs.python.org/3.12/library/exceptions.html#DeprecationWarning) or a [`UserWarning`](https://docs.python.org/3.12/library/exceptions.html#UserWarning). While the former is more descriptive, it is hidden by default, so the latter might be a better choice. In addition, I would recommend, as you suggested, making the camelCase functions into wrappers for the snake_case functions. That way, you can also interleave the new function definitions, avoiding the need to do any copying of the function bodies.

Here is an example of what I'm suggesting:

```suggestion
    def getSimplificationsUsed(self):
        warnings.warn(
            f"You are using the camel case naming convention ({self.getSimplificationsUsed.__name__!r}) for the Python API. "
            f"Please use snake case ({self.get_simplifications_used.__name__!r}) instead.",
            DeprecationWarning,
        )
        self.get_simplifications_used()
    
    def get_simplifications_used(self):
        return self.server.getSimplificationsUsed()
```

It's not perfect and will requires some tweaking. For example, you will need to use the wrapper to convert the style of parameter names as well.

As a side note, instead of screenshots of code, you should use [Markdown code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks). They allow copying and pasting, and GitHub automatically adds syntax highlighting if you declare the language.

Reply:
This sounds good. Thanks for the feedback! I will go ahead an implement today. I was wondering about the DeprecationWarning vs UserWarning. It seems weird to have one of them hidden by default, since the default logging level is warning. Oh well

Reply:
> Thanks for the feedback!

You're welcome! When using this library, it definitely bothered me that it wasn't PEP 8 compliant, so I'm glad this change is happening.

> I was wondering about the DeprecationWarning vs UserWarning. It seems weird to have one of them hidden by default, since the default logging level is warning. Oh well

From what I've seen, even some of the core Python developers find it a little problematic. Ideally, you are supposed to test with `DeprecationWarning`s visible, but I (and most other people) usually forget to do that.

Reply:
I made the changes. LMK if anything else needs fixes.

Reply:
I didn't do a full review, but I skimmed it over and it looks good to me!

Reply:
Thanks @aphedges. We can iterate quickly if anything breaks. Hopefully having Github action will help with that in the future.